libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -217,7 +217,10 @@ func (p *Provider) emitEvents(pod *kubernetes.Pod, flag string, containers []kub
 
 		// This must be an id that doesn't depend on the state of the container
 		// so it works also on `stop` if containers have been already deleted.
-		eventID := fmt.Sprintf("%s.%s", pod.GetObjectMeta().GetUID(), c.Name)
+		eventID := fmt.Sprintf("%s", pod.GetObjectMeta().GetUID())
+		if p.bus.GetName() != "metricbeat" {
```

Comment:
I would avoid this kind of ifs, if at some point in the future we want to use autodiscover monitor network endpoints in other beat (for example with heartbeat), we can get crazy finding why it behaves differently in metricbeat.

If we decide to go on with this solution as a quick fix for #12011 please create a follow-up issue to find a longer term solution.

Reply:
+1 on avoiding beat specific checks on libbeats.